File: C#/IsPalindrome.cs

using System.Text;
//O(n)
//O(n)
public class Solution
{
    public bool IsPalindrome(string s)
    {
        if (string.IsNullOrEmpty(s)) return false;

        var cleanedString = new StringBuilder();

        foreach (char c in s)
        {
            if (char.IsLetterOrDigit(c))
                cleanedString.Append(char.ToLower(c));
        }

        string cleaned = cleanedString.ToString();

        int L = 0, R = cleaned.Length - 1;

        while (L < R)
        {
            if (cleaned[L] == cleaned[R])
            {
                L++;
                R--;
            }
            else
            {
                return false;
            }
        }

        return true;
    }
}

File: C++/isPalindrome.cpp
// Time: O(N)
// Space: O(N)
#include <string>

class Solution
{
public:
    bool isPalindrome(int x)
    {
        if (x == 0)
            return true;
        if (x < 0)
            return false;

        auto num = std::to_string(x);
        int L = 0, R = num.size() - 1;

        while (L < R)
        {
            if (num[L] != num[R])
                return false;
            L++;
            R--;
        }

        return true;
    }
};